src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -601,18 +603,38 @@ private SupportedPlatform GetCurrentRunningOS()
                     context.TargetUri.GetFileName()));
             }
 
-            if (options.ComputeFileHashes && _resultsCachingLogger.HashToResultsMap.TryGetValue(context.Hashes.Sha256, out List<Tuple<ReportingDescriptor, Result>> results))
+            if (options.ComputeFileHashes)
             {
-                context.Logger.AnalyzingTarget(context);
-                foreach (Tuple<ReportingDescriptor, Result> result in results)
+                _resultsCachingLogger.HashToResultsMap.TryGetValue(context.Hashes.Sha256, out List<Tuple<ReportingDescriptor, Result>> results);
+                _resultsCachingLogger.HashToNotificationsMap.TryGetValue(context.Hashes.Sha256, out List<Notification> notifications);
+
+                bool replayCachedData = (results != null || notifications != null);
+
+                if (replayCachedData)
                 {
-                    if (result.Item2.Locations?.Count > 0)
+                    context.Logger.AnalyzingTarget(context);
+
+                    if (results != null)
+                    {
+                        foreach (Tuple<ReportingDescriptor, Result> result in results)
+                        {
+                            if (result.Item2.Locations?.Count > 0)
+                            {
+                                result.Item2.Locations[0].PhysicalLocation.ArtifactLocation.Uri = context.TargetUri;
```

Comment:
>TargetUri [](start = 106, length = 9)

Maybe this is the answer to my question? Can the same dependent binary and/or PDB occur in more than one place, if we're analyzing a set of binaries?

Reply:
Yes, that's it, I see now.

---
In reply to: [316890439](https://github.com/microsoft/sarif-sdk/pull/1658#discussion_r316890439) [](ancestors = 316890439)

Reply:
Indeed. Imagine a build out of MyDll, MyOtherDll, and MyExe, all of which have a dependency on Newtonsoft.Json. The build system will copy that dependency to all three locations. The scan automation points at the high level output directly and ends up analyzing all three copies. 

---
In reply to: [316897163](https://github.com/microsoft/sarif-sdk/pull/1658#discussion_r316897163) [](ancestors = 316897163,316890439)